airbot_error_manager:
  ros__parameters:
    log_print: true
    # 0: unset, 10: debug, 20: info, 30: warn, 40: error
    log_level: 20
    publish_rate_ms: 1000
    error_publish_cnt: 1 # 에러 발생/해제 메시지 퍼블리싱 횟수 (to UDP_interface)
    error_list_size: 28 # 관리할 에러 리스크 최대 크기
    error_list: error_list.yaml

airbot_error_monitor:
  ros__parameters:
    low_battery_error:
      monitoring_rate_ms: 1000
      occure:
        battery_percentage_min: 10
        battery_percentage_max: 15
      release:
        battery_percentage_th: 20
        duration_sec: 30.0

    discharging_error:
      monitoring_rate_ms: 1000
      occure:
        battery_percentage_min: 0
        battery_percentage_max: 10
        duration_sec: 10.0
      release:
        battery_percentage_th: 15
        duration_sec: 30.0

    board_overheat_error:
      monitoring_rate_ms: 1000
      occure:
        temperature_th_c: 70
        duration_sec: 30.0

    charging_error:
      monitoring_rate_ms: 1000
      occure:
        battery_percentage_th: 60
        duration_sec: 1200.0

    fall_down_error:
      monitoring_rate_ms: 1000
      occure:
        drop_ir_adc_th: 900
        drop_ir_cnt_min: 3
        imu_roll_th_deg: 60
        imu_pitch_th_deg: 60

    lift_error:
      monitoring_rate_ms: 10
      occure:
        drop_ir_adc_th: 900
        drop_ir_cnt_min: 4
        imu_z_acc_low_th: 9.2
        imu_z_acc_hight_th: 10.5

    cliff_error:
      monitoring_rate_ms: 10
      occure:
        duration_sec: 3.0 # 사용 X
        accum_dist_th_m: 0.3
